protocols/relay/src/behaviour/handler.rs

```diff
@@ -879,9 +879,7 @@ impl ConnectionHandler for Handler {
         if self.reservation_request_future.is_none()
             && self.circuit_accept_futures.is_empty()
             && self.circuit_deny_futures.is_empty()
-            && self.alive_lend_out_substreams.is_empty()
             && self.circuits.is_empty()
-            && self.active_reservation.is_none()
```

Comment:
This is wrong. In fact, the only condition that should remain here is whether we have an active reservation! :)

Reply:
fixed 5db7968